src/NHibernate/Transform/CacheableResultTransformer.cs

```diff
@@ -152,13 +172,24 @@ private CacheableResultTransformer(bool[] includeInTuple, bool[] includeInTransf
 			}
 		}
 
+		internal void SupplyAutoDiscoveredParameters(IResultTransformer transformer, string[] aliases)
+		{
+			if (!AutoDiscoverTypes)
+				throw new InvalidOperationException(
+					"Cannot supply auto-discovered parameters when it is not enabled on the transformer.");
+
+			var includeInTuple = ArrayHelper.Fill(true, aliases?.Length ?? 0);
+			InitializeTransformer(includeInTuple, GetIncludeInTransform(transformer, aliases, includeInTuple));
+		}
 
 		public object TransformTuple(object[] tuple, string[] aliases)
 		{
-			if (aliases != null && aliases.Length != _tupleLength)
+			if (_includeInTuple == null)
+				throw new InvalidOperationException("This transformer is not initialized");
+			if (aliases != null && aliases.Length != _includeInTuple.Length)
 			{
 				throw new InvalidOperationException(
-					"aliases expected length is " + _tupleLength +
+					"aliases expected length is " + _includeInTuple.Length +
```

Comment:
This change is questioning. Tests succeeds, but why should it be so? Moreover it appears here that `aliases` are only used for causing this failure. The actual implementation does not need them. Maybe this check should be removed instead.